File: MADVCamera/CameraManager/ambaresponse/AMBAFileTransferResultResponse.m
 //
//  AMBAFileTransferResultResponse.m
//  Madv360_v1
//
//  Created by DOM QIU on 16/9/16.
//  Copyright © 2016年 Cyllenge. All rights reserved.
//

#import "AMBAFileTransferResultResponse.h"

@implementation AMBAFileTransferResultResponse

- (NSString*) md5 {
    NSArray* paramArray = (NSArray*) self.param;
    NSDictionary* md5Dict = (NSDictionary*) paramArray[1];
    NSString* md5 = (NSString*) md5Dict[@"md5sum"];
    return md5;
}

- (NSInteger) bytesSent {
    NSArray* paramArray = (NSArray*) self.param;
    NSDictionary* bytesSentDict = (NSDictionary*) paramArray[0];
    double bytesSent = [bytesSentDict[@"bytes sent"] doubleValue];
    return (NSInteger)bytesSent;
}

- (NSInteger) bytesReceived {
    NSArray* paramArray = (NSArray*) self.param;
    NSDictionary* bytesReceivedDict = (NSDictionary*) paramArray[0];
    double bytesReceived = [bytesReceivedDict[@"bytes received"] doubleValue];
    return (NSInteger)bytesReceived;
}

@end

File: MADVCamera/CameraManager/ambaresponse/AMBAFileTransferResultResponse.java
package com.madv360.madv.connection.ambaresponse;

import com.madv360.madv.connection.AMBAResponse;

import java.util.ArrayList;
import java.util.Map;

/**
 * Created by qiudong on 16/6/25.
 */
public class AMBAFileTransferResultResponse extends AMBAResponse {
    public String getMD5() {
        ArrayList paramArray = (ArrayList) param;
        Map md5Map = (Map) paramArray.get(1);
        String md5 = (String) md5Map.get("md5sum");
        return md5;
    }

    public long getBytesSent() {
        ArrayList paramArray = (ArrayList) param;
        Map bytesSentMap = (Map) paramArray.get(0);
        Double dBytesSent = (Double) bytesSentMap.get("bytes sent");
        double bytesSent = (null == dBytesSent ? 0 : dBytesSent);
        return (long) bytesSent;
    }

    public long getBytesReceived() {
        ArrayList paramArray = (ArrayList) param;
        Map bytesReceiveMap = (Map) paramArray.get(0);
        Double dBytesReceive = (Double) bytesReceiveMap.get("bytes received");
        double bytesReceive = (null == dBytesReceive ? 0 : dBytesReceive);
        return (long) bytesReceive;
    }
}
